### PR TITLE
Fixing win_perf_counters tests

### PR DESCRIPTION
**Pull request by JeffAshton**
Addresses issue https://github.com/influxdata/telegraf/issues/3062

Minimal changes to get the existing unit tests compiling and passing again. 

### Required for all PRs:

- [Y] Signed [CLA](https://influxdata.com/community/cla/).
- [N/A] Associated README.md updated.
- [Y] Has appropriate unit tests.

### CODE REVIEW

**Comment by JeffAshton** [comment]
Appveyor build with these changes:

https://ci.appveyor.com/project/Brightspace/telegraf/build/1.4-rc102

https://github.com/Brightspace/telegraf/commits/fixing-win-perf-counter-tests-with-appveyor

File: plugins/inputs/win_perf_counters/win_perf_counters_test.go [hunk 1]
```diff
@@ -435,6 +435,8 @@ func TestWinPerfcountersCollect1(t *testing.T) {
 	instances[0] = "_Total"
 	counters[0] = "Parking Status"
 
+	var expectedCounter string = "Parking_Status"
```
Comment (JeffAshton) [inline plugins/inputs/win_perf_counters/win_perf_counters_test.go]: These tests were broken as a part of this change set for Sanitize Field name: https://github.com/influxdata/telegraf/commit/015261a524e8f335f844e7e3ffdd75ea3b4f2b57

File: plugins/inputs/win_perf_counters/win_perf_counters_test.go [hunk 2]
```diff
@@ -169,36 +169,37 @@ func TestWinPerfcountersConfigGet4(t *testing.T) {
...
-	metrics := itemList{}
 
 	var instances = make([]string, 2)
 	var counters = make([]string, 2)
 	var perfobjects = make([]perfobject, 1)
 
 	objectname := "Processor Information"
 	instances[0] = "_Total"
-	instances[1] = "0"
+	instances[1] = "0,1"
```
Comment (JeffAshton) [inline plugins/inputs/win_perf_counters/win_perf_counters_test.go]: Later versions of windows have multi-core support.

image

File: plugins/inputs/win_perf_counters/win_perf_counters_test.go [hunk 3]
```diff
@@ -12,7 +12,6 @@ import (
 )
 
 func TestWinPerfcountersConfigGet1(t *testing.T) {
-	validmetrics := itemList{}
```
Comment (JeffAshton) [inline plugins/inputs/win_perf_counters/win_perf_counters_test.go]: itemList was removed in this pull request https://github.com/influxdata/telegraf/pull/2800

File: plugins/inputs/win_perf_counters/win_perf_counters_test.go [hunk 4]
```diff
@@ -39,14 +38,13 @@ func TestWinPerfcountersConfigGet1(t *testing.T) {
 
 	perfobjects[0] = PerfObject
 
-	m := Win_PerfCounters{PrintValid: false, TestName: "ConfigGet1", Object: perfobjects}
```
Comment (JeffAshton) [inline plugins/inputs/win_perf_counters/win_perf_counters_test.go]: TestName was also removed in this pull request https://github.com/influxdata/telegraf/pull/2800

File: plugins/inputs/win_perf_counters/win_perf_counters_test.go [hunk 5]
```diff
@@ -73,26 +71,27 @@ func TestWinPerfcountersConfigGet2(t *testing.T) {
...
 
-	m := Win_PerfCounters{PrintValid: false, TestName: "ConfigGet2", Object: perfobjects}
+	m := Win_PerfCounters{PrintValid: false, Object: perfobjects}
 
-	err := m.ParseConfig(&metrics)
+	err := m.ParseConfig()
 	require.NoError(t, err)
 
-	if len(metrics.items) == 1 {
+	var parsedItems = m.GetParsedItemsForTesting()
```
Comment (JeffAshton) [inline plugins/inputs/win_perf_counters/win_perf_counters_test.go]: Added GetParsedItemsForTesting to satisfy existing tests without too drastic a rewrite

**Comment by danielnelson** [comment]
Thank you

**Pull request merged**